plugin/src/main/java/org/opensearch/ml/action/memorycontainer/memory/MemoryProcessingService.java

```diff
@@ -317,4 +365,42 @@ private List<MemoryDecision> parseMemoryDecisions(MLTaskResponse response) {
             throw new RuntimeException("Failed to parse memory decisions", e);
         }
     }
+
+    public void summarizeMessages(List<MessageInput> messages, ActionListener<String> listener) {
+        if (messages == null || messages.isEmpty()) {
+            listener.onResponse("");
```

Comment:
Is it expected to response an empty String or should an exception to be thrown?
throw new IllegalStateException("Messages list cannot be empty");